walk-in-interview/src/main/webapp/account/firebase-auth.js

```diff
@@ -92,67 +80,112 @@ Auth.addPhoneAuthUI = (elementId, successPath) => {
- *
- * @param {String} phoneNumber The phone number for the new applicant account.
- * @param {Object} appVerifier The recaptcha verifier.
- */
-Auth.createApplicantAccount = (phoneNumber, appVerifier) => {
-  // TODO(issue/79): set up phone number sign in with otp and recaptcha
-};
-
-/**
- * This will sign into an existing applicant account.
- *
- * @param {String} phoneNumber The number for the existing applicant account.
- * @param {Object} appVerifier The recaptcha verifier.
- */
-Auth.signIntoApplicantAccount = (phoneNumber, appVerifier) => {
-  // TODO(issue/79): set up phone number sign in with otp and recaptcha
-};
-
 /**
  * Signs out the current user.
- *
- * @param {String} elementId The div in which to show the signed out status.
  */
-Auth.signOutCurrentUser = (elementId) => {
+Auth.signOutCurrentUser = () => {
   firebase.auth().signOut().then(() => {
     console.log('sign out successful');
-    document.getElementById(elementId).innerText = STRINGS['sign-out-success'];
+    // TODO(issue/100): set the cookie at the server side instead
+    setCookie(USER_TYPE_COOKIE_PARAM, TYPE_NO_USER);
+    alert(STRINGS['sign-out-success']);
   }).catch((error) => {
     console.error(error);
   });
+  Auth.subscribeToUserAuthenticationChanges();
 };
 
 /**
  * Checks the user sign in status.
+ * Makes a POST request to create a session cookie when the user signs in.
+ * Makes a POST request to clear the session cookie when the user signs out.
+ */
+Auth.subscribeToUserAuthenticationChanges = () => {
+  firebase.auth().onAuthStateChanged((firebaseUser) => {
+    // User not signed in.
+    if (!firebaseUser) {
+      console.log('User Not Signed In');
+      // Clears the session cookie
+      Auth.postIdTokenToSessionLogout(API['log-out'])
+          .catch((error) => {
+            console.log(error);
+          });
+      return;
+    }
+      
+    // User signed in.
+    console.log('User Signed In');
+    // Get the user's ID token as it is needed to exchange
+    // for a session cookie.
+    firebaseUser.getIdToken()
+        .then((idToken) => {
+          // Session login endpoint is queried and session cookie is set.
+          // CSRF protection should be taken into account.
+          const csrfToken = getCookie('csrfToken');
+
+          if (idToken == localStorage.getItem('idToken') && csrfToken == localStorage.getItem('csrfToken')) {
+            return;
+          }
+
+          if (idToken != localStorage.getItem('idToken')) {
+            localStorage.setItem('idToken', idToken);
+          }
+
+          if (csrfToken != localStorage.getItem('csrfToken')) {
+            localStorage.setItem('csrfToken', csrfToken);
+          }
+
+          return Auth.postIdTokenToSessionLogin(API['log-in'],
+              idToken, csrfToken);
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+  });
+};
+
+/**
+ * Checks the current user status.
  */
 Auth.checkCurrentUser = () => {
```

Comment:
Now that we have subscribeToUserAuthenticationChanges, what is this method used for?

Reply:
+1 i'm not sure also